### PR TITLE
Use gulp to watch files/livereload

### PR DESCRIPTION
**Pull request by maxlang**
The main reason to use Gulp is Livereload, which makes it a lot easier to make visual changes.

I've also switched to use Bower to manage frontend dependencies and tsd to manage Typescript definitions. The exception is the mithril.d.js file, which we've made some modifications to.

This shouldn't add any extra dependencies except for when developing.

### CODE REVIEW

**Comment by maxlang** [comment]
I'll add sourcemaps/non-minified versions for debugging in a future commit. I'll also try to make a PR against https://github.com/borisyankov/DefinitelyTyped to get our version of mithril type defnitions up since it's on 0.10 (and @mrtracy mentioned there were bugs in the 0.20 version)

File: ui/README.md [hunk 1]
```diff
@@ -26,6 +28,19 @@ Before committing, be sure to run `make` to generate a non-debug version of
 `embedded.go`. This is enforced by our build system, but forgetting to do this
 will result in wasted time waiting for the build.
 
+## Watch/Livereload
+If you want to use gulp watch, you'll need to install gulp globally
+([see dependencies](####dependencies).) Then use `gulp` to start auto-recompiling
```
Comment (tschottdorf) [inline ui/README.md]: the `####` should go.

File: ui/README.md [hunk 2]
```diff
@@ -34,3 +49,14 @@ typescript compiler locally. We also require a typescript linter (tslint).
 2. Install [typescript](http://www.typescriptlang.org/) and
    [tslint](https://github.com/palantir/tslint). This can be done via the
    command line using npm: `npm install -g typescript tslint`
+
+We use bower to manage frontend dependencies and tsd to manage typescript
+definition files. We also use gulp as task runner and for livereload.
+These will be automatically run by make, but if you want to run them yourself
+you'll need to install them globally using npm: `npm install -g bower tsd gulp`
```
Comment (tschottdorf) [inline ui/README.md]: is `-g` really needed to run them?

File: ui/README.md (hunk 1 above)
Comment (mrtracy) [inline ui/README.md]: Aren't the #'s necessary in order to make a proper internal link?  It's a github extension to markdown.

Comment (tschottdorf) [inline ui/README.md]: this is a normal link, and if you want to link to the element with `id="dependencies"`, then you should link to `#dependencies`. So above I really wanted to say `s/####/#/`.

**Comment by maxlang** [comment]
Assigned @tamird because we worked on this together. @mrtracy you might want to take a look as well.

**Comment by maxlang** [comment]
Whoops @tamird I remade the commits with the trailing newline and lost your comments

**Comment by mrtracy** [comment]
LGTM, this is looking much better.

**Comment by tamird** [comment]
LGTM couple comments left on commits

**Comment by maxlang** [comment]
@tamird and @mrtracy - some final changes. Notably I now set permissions/delete index.html correctly in the gulp file and I correctly generate app_debug.css in the gulp file.

**Comment by tamird** [comment]
:+1:

**Pull request merged**